File: 조준장/골드5/14719.py
import sys

h, w = map(int, sys.stdin.readline().split(" "))
bloc_h = list(map(int, sys.stdin.readline().split(" ")))
answer = 0

for i in range(1, w - 1):
    left = max(bloc_h[:i])
    right = max(bloc_h[i + 1:])

    compare = min(left, right)

    if bloc_h[i] < compare:
        answer += compare - bloc_h[i]

print(answer)

File: 조준장/골드5/14719.kt
import kotlin.math.min

class `14719` {
    fun solve() {
        val br = System.`in`.bufferedReader()
        val bw = System.out.bufferedWriter()
        val sb = StringBuilder()

        val (h, w) = br.readLine().split(" ").map { it.toInt() }
        val arr = br.readLine().split(" ").map { it.toInt() }

        var answer = 0
        for (i in 1 until w) {
            val left = arr.subList(0, i).maxOrNull() ?: 0
            val right = arr.subList(i + 1, w).maxOrNull() ?: 0

            val compare = min(left, right)

            if (arr[i] < compare) {
                answer += compare - arr[i]
            }
        }

        sb.append(answer)
        bw.write(sb.toString())

        br.close()
        bw.close()
    }
}

fun main() {
    `14719`().solve()
}
